### ISSUE TITLE
Numeric precision unit test failure on 32 bit

### ISSUE THREAD

**Issue by jeroenooms** [lme4/lme4#372]
As part of the migration to the new windows toolchain, we found that one unit tests is failing on some architectures. CRAN is concerned about adopting the new windows toolchain is lme4 is not passing checks, however it looks like this problem is not directly related to windows.

Is it expected that numeric precision is less accurate on 32bit architectures? In that case would it make sense to relax the unit test?

```
1. Failure (at test-nbinom_refit.R#18): glmer refit ----------------------------
m.base@beta not equal to (m.base.r <- refit(m.base))@beta
6/6 mismatches (average diff: 5.67e-06).
First 6:
 pos      x      y      diff
   1  1.631  1.631  4.46e-06
   2  0.767  0.767 -6.90e-06
   3  1.011  1.011 -6.07e-06
   4  1.512  1.512 -2.50e-06
   5 -0.615 -0.615  7.28e-06
   6 -0.610 -0.610  6.84e-06
```

This problem has been confirmed on:

  - gcc 4.9.3 (win32)
  - gcc-4.4.7 (CentOS 6, i386)
  - solaris Sparc, see [cran log](https://www.r-project.org/nosvn/R.check/r-patched-solaris-sparc/lme4-00check.html) 

Smaller reproducible example from @jimhester.

```r
set.seed(101)
dd <- expand.grid(f1 = factor(1:3),
                  f2 = LETTERS[1:2], g=1:9, rep=1:15,
          KEEP.OUT.ATTRS=FALSE)
mu <- 5*(-4 + with(dd, as.integer(f1) + 4*as.numeric(f2)))
dd$y <- rnbinom(nrow(dd), mu = mu, size = 0.5)

m.base <- glmer(y ~ f1*f2 + (1|g), data=dd, family=poisson)
m.base.r <- refit(m.base)
all.equal(m.base@beta, m.base.r@beta)
#> [1] "Mean relative difference: 5.537048e-06"
```

**Comment by bbolker** [comment]
**tl;dr** I'm not surprised.

I don't personally have any objection to relaxing the precision to 1e-5.  Confession: due to a combination of organic code growth (ugh) and the fact that we are layering nonlinear optimization on top of fairly complicated numerical linear algebra, we don't have a principled way of deciding what level of mismatch would actually constitute something going wrong with the algorithm ...

**Comment by jeroenooms** [comment]
OK so I guess we should ignore this problem. Would it be possible to push out a bugfix release? CRAN is  currently considering adopting the new toolchain for the R 3.3 release but they are concerned if "very important packages" (their words) are not passing checks....

**Comment by bbolker** [comment]
Not impossible.  It's been a few months since our last release.  Of course I'm frustrated that we haven't made more progress/addressed more long-standing issues, but that's just the usual regrets. I will have to run all the downstream package checks, but hopefully I have the infrastructure for that a little better set up ...

**Comment by jeroenooms** [comment]
Actually there is another check that fails due to tolerance (it didn't show up at first because the tests stop at the first failure....)

**Comment by bbolker** [comment]
Would be good to know what it is, since I don't have access to a 32-bit Windows test system ... Or you could submit a pull request.

**Comment by jeroenooms** [comment]
Here you go: https://github.com/lme4/lme4/pull/373. This seems to fix all issues on my machine. But let's wait for @jimhester to confirm before pushing out a release...

**Comment by mmaechler** [comment]
Late coming, let me confirm that I share @bbolker 's feelings, and indeed, we really should be good citizens and help with the new toolchain making it into R with as little CRAN ripple effects as possible.
Thank you Jeroen and Ben!

**Comment by jeroenooms** [comment]
I think this is good to go. If we could schedule out a release anytime soon-ish that takes us one small step closer to the new windows toolchain in R 3.3.

**Comment by jimhester** [comment]
Sorry for the delay, I can confirm that all tests pass with the current master (23771d81 v1.1.-12) on Windows R 32 bit.  As Jeroen said doing a bugfix release as soon as possible would be appreciated.

Thanks again for looking at this and the fix.

**Comment by bbolker** [comment]
OK.  I will close this.  I have to fight with the reverse dependency checks (there are glitches every time, it seems), then I will submit.
